### PR TITLE
Add methods to convert objects to JSON text

### PR DESCRIPTION
**Pull request by bluedreamer**
Added a switch "-j" to vme to dump stuff as json
Added new command "jstat" to output stats in json format

### CODE REVIEW

File: vme/src/combat.cpp [hunk 1]
```diff
@@ -543,6 +543,37 @@ void cCombat::status(const unit_data *god)
...
+        writer.Int(nAttackNo);
+
+        writer.String("When");
+        writer.Int(nWhen);
+
+        json::write_pointer_kvp("Owner", pOwner, writer);
+
+        json::write_pointer_kvp("Melee", pMelee, writer);
+
+        writer.String("NoOpponents");
```
Comment (Seifert69) [inline vme/src/combat.cpp:560]: Most of these are in-game variables, how do I differentiate these?

File: entrypoint.sh [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+  if [ $done -eq 0 ]; then
+    kill $(pgrep vme)
+    kill $(pgrep mplex)
+  fi
+  done=1
+}
+
+trap signals INT TERM EXIT HUP
+
+service nginx restart
```
Comment (Seifert69) [inline entrypoint.sh:16]: Is entrypoint only for docker? If not, isn't it kind of risky to restart nginx?

File: vme/src/db_file.cpp [hunk 3]
```diff
@@ -1173,11 +1180,8 @@ void bwrite_block(FILE *datafile, int length, void *buffer)
 int write_unit_string(CByteBuffer *pBuf, unit_data *u)
 {
     int i = 0;
-
-#ifdef DMSERVER
```
Comment (Seifert69) [inline vme/src/db_file.cpp:1177]: Suppose I'm wondering about this line. Presumably it was there related to something with VMC... ?

File: vme/src/file_index_type.cpp [hunk 4]
```diff
@@ -191,6 +190,11 @@ zone_type *file_index_type::getZone() const
     return m_zone;
 }
 
+std::string file_index_type::getID() const
```
Comment (Seifert69) [inline vme/src/file_index_type.cpp:193]: We've called this the symbolic name / aka .symname in DIL.

File: entrypoint.sh (hunk 2 above)
Comment (bluedreamer) [reply entrypoint.sh:16]: Docker only

File: vme/src/db_file.cpp (hunk 3 above)
Comment (bluedreamer) [reply vme/src/db_file.cpp:1177]: Yeah, I think they were to reduce compile times. Adding some of the json_helper stuff to mplex dragged in almost everything else. There is a lot of dependencies between translation units :-)

File: vme/src/file_index_type.cpp (hunk 4 above)
Comment (bluedreamer) [reply vme/src/file_index_type.cpp:193]: Ah - should probably use that then instead

**Comment by Seifert69** [comment]
Sorry, was moving :-)

**Pull request merged**